File: Java/src/main/java/com/adaptionsoft/games/trivia/runner/GameRunner.java

package com.adaptionsoft.games.trivia.runner;
import java.util.Random;

import com.adaptionsoft.games.uglytrivia.Action;
import com.adaptionsoft.games.uglytrivia.Game;


public class GameRunner {

	private static boolean notAWinner;

	public static void main(String[] args) {
	    Random rand = new Random();
	    main(rand);
	}

	static void main(Random rand) {
		Game aGame = new Game();
		
		aGame.getAdd("Chet").execute();
		aGame.getAdd("Pat").execute();
		aGame.getAdd("Sue").execute();
		
	
		do {
			
			Action roll = aGame.getRoll().create(rand);
			roll.execute();
			
			Action answer = aGame.getAnswer().create(rand);
			notAWinner = answer.execute();
			
			
		} while (notAWinner);
		
	}
}

File: Csharp/Trivia/GameRunner.cs
using System;
using UglyTrivia;

namespace Trivia
{
    public class GameRunner
    {
        private static bool notAWinner;

        public static void Main(String[] args)
        {
            System.Random rand = new System.Random();
            Run(rand);
        }

        public static void Run(System.Random rand)
        {
            Game aGame = new Game();

            aGame.GetAdd("Chet").Execute();
            aGame.GetAdd("Pat").Execute();
            aGame.GetAdd("Sue").Execute();

            do
            {

                UglyTrivia.Action roll = aGame.GetRoll().Create(rand);
                roll.Execute();

                UglyTrivia.Action answer = aGame.GetAnswer().Create(rand);
                notAWinner = answer.Execute();



            } while (notAWinner);

        }


    }

}
